lnencrypt/tor.go

```diff
@@ -0,0 +1,76 @@
+package lnencrypt
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+var (
+	// ErrEncryptedTorPrivateKey is thrown when a tor private key is
+	// encrypted, but the user requested an unencrypted key.
+	ErrEncryptedTorPrivateKey = errors.New("it appears the Tor private key " +
+		"is encrypted but you didn't pass the --tor.encryptkey flag. " +
+		"Please restart lnd with the --tor.encryptkey flag or delete " +
+		"the Tor key file for regeneration")
+)
+
+// ReadTorPrivateKey reads the Tor private key from disk and returns the
+// contents. If the file is encrypted it also decrypts the file
+// before returning.
+func ReadTorPrivateKey(keyPath string, encryptKey bool, keyRing keychain.KeyRing) (
```

Comment:
Not sure if I'm missing something -- I don't think these read/write functions are needed, this is already taken care of by the `OnionFile`, why not add encryption support directly to that instead? Doing so would prevent some of the existing changes to `AddOnion`.

Reply:
I think this will make the `tor` package depend on `lnencrypt` if it must encrypt/decrypt?

Reply:
Correct. If we want to avoid that, which I think we should, we'd have to move the `OnionFile` implementation out of the package to somewhere else (lnd package?).

Reply:
That sounds like a better approach

Reply:
Looking through these comments, since I'll be taking over this PR for Graham. It looks like if we move OnionFile to lnd package, I think that will create a circular dependency when OnionFile is used in the watchtower package. So I'm leaning towards moving OnionFile to a new package... Does that sound reasonable?

Reply:
@orbitalturtle that also works, could you make it a subpackage of the existing tor pacakge?